core/eolearn/core/eodata.py

```diff
@@ -254,100 +231,104 @@ class EOPatch:
+        self.bbox = bbox
+        self.timestamps = timestamps or []
 
     @property
     def timestamp(self) -> List[dt.datetime]:
-        """A property for handling the deprecated timestamp attribute.
-
-        :return: A list of EOPatch timestamps
-        """
-        warn(
-            "The attribute `timestamp` is deprecated, use `timestamps` instead.",
-            category=EODeprecationWarning,
-            stacklevel=2,
-        )
+        """A property for handling the deprecated timestamp attribute."""
+        warn(TIMESTAMP_RENAME_WARNING, category=EODeprecationWarning, stacklevel=2)
         return self.timestamps
 
     @timestamp.setter
     def timestamp(self, value: List[dt.datetime]) -> None:
-        warn(
-            "The attribute `timestamp` is deprecated, use `timestamps` instead.",
-            category=EODeprecationWarning,
-            stacklevel=2,
-        )
+        warn(TIMESTAMP_RENAME_WARNING, category=EODeprecationWarning, stacklevel=2)
         self.timestamps = value
 
-    def __setattr__(self, key: str, value: object, feature_name: Union[str, None, EllipsisType] = None) -> None:
-        """Raises TypeError if feature type attributes are not of correct type.
-
-        In case they are a dictionary they are cast to _FeatureDict class.
-        """
-        if feature_name not in (None, Ellipsis) and FeatureType.has_value(key):
-            self.__getattribute__(key)[feature_name] = value
-            return
-
-        if FeatureType.has_value(key) and not isinstance(value, FeatureIO):
-            feature_type = FeatureType(key)
-            value = self._parse_feature_type_value(feature_type, value)
-
-        super().__setattr__(key, value)
+    @property
+    def timestamps(self) -> List[dt.datetime]:
+        """A property for handling the `timestamps` attribute."""
+        return self._timestamps
+
+    @timestamps.setter
+    def timestamps(self, value: Iterable[dt.datetime]) -> None:
+        if isinstance(value, Iterable) and all(isinstance(time, (dt.date, str)) for time in value):
+            self._timestamps = [self._parse_to_datetime(time) for time in value]
+        else:
+            raise TypeError(f"Cannot assign {value} as timestamps. Should be a sequence of datetime.datetime objects.")
 
     @staticmethod
-    def _parse_feature_type_value(
-        feature_type: FeatureType, value: object
-    ) -> Union[_FeatureDict, BBox, List[dt.date], None]:
-        """Checks or parses value which will be assigned to a feature type attribute of `EOPatch`. If the value
-        cannot be parsed correctly it raises an error.
-
-        :raises: TypeError, ValueError
-        """
-
-        if feature_type is FeatureType.BBOX and (value is None or isinstance(value, BBox)):
-            if value is None:
-                warn(MISSING_BBOX_WARNING, category=EODeprecationWarning, stacklevel=2)
-            return value
+    def _parse_to_datetime(value: Union[dt.date, str]) -> dt.datetime:
```

Comment:
Perhaps the `parse_time` from sh-py could/should be used here?

Reply:
Splendid idea, not sure how i missed it :see_no_evil: